test/test_metalearning/pyMetaLearn/test_meta_features.py

```diff
@@ -16,475 +19,598 @@
+    )
+    meta_features.helper_functions.set_value(
+        "ClassOccurences",
+        meta_features.helper_functions["ClassOccurences"](X, y, logger),
+    )
+    if request.param == 'numpy':
+        return X, y, categorical
+    elif request.param == 'pandas':
+        return pd.DataFrame(X), y, categorical
+    else:
+        raise ValueError(request.param)
+
+
+@pytest.fixture(
+    scope='class',
+    params=('pandas', 'numpy')
+)
+def meta_train_data_transformed(request):
+    tests_dir = __file__
+    os.chdir(os.path.dirname(tests_dir))
+
+    decoder = arff.ArffDecoder()
+    with open(os.path.join("datasets", "dataset.arff")) as fh:
+        dataset = decoder.decode(fh, encode_nominal=True)
+
+    # -1 because the last attribute is the class
+    attribute_types = [
+        'numeric' if type(type_) != list else 'nominal'
+        for name, type_ in dataset['attributes'][:-1]]
+    categorical = [True if attribute == 'nominal' else False
+                   for attribute in attribute_types]
+
+    data = np.array(dataset['data'], dtype=np.float64)
+    X = data[:, :-1]
+    y = data[:, -1].reshape((-1,))
+
+    logger = logging.getLogger('Meta')
+    meta_features.helper_functions.set_value(
+        "MissingValues", meta_features.helper_functions["MissingValues"](X, y, logger, categorical),
+        )
+    meta_features.helper_functions.set_value(
+        "NumSymbols",
+        meta_features.helper_functions["NumSymbols"](X, y, logger,  categorical),
+    )
+    meta_features.helper_functions.set_value(
+        "ClassOccurences",
+        meta_features.helper_functions["ClassOccurences"](X, y, logger),
+    )
+
+    DPP = DataPreprocessor(feat_type={
+        i: 'categorical' if feat else 'numerical' for i, feat in enumerate(categorical)
+    })
+    X_transformed = DPP.fit_transform(X)
+
+    number_numerical = np.sum(~np.array(categorical))
+    categorical_transformed = [True] * (X_transformed.shape[1] -
+                                        number_numerical) + \
+                              [False] * number_numerical
+    categorical_transformed = categorical_transformed
+
+    # pre-compute values for transformed inputs
+    meta_features.helper_functions.set_value(
+        "PCA", meta_features.helper_functions["PCA"](X_transformed, y, logger),
+    )
+    meta_features.helper_functions.set_value(
+        "Skewnesses", meta_features.helper_functions["Skewnesses"](
+            X_transformed, y, logger, categorical_transformed),
+    )
+    meta_features.helper_functions.set_value(
+        "Kurtosisses", meta_features.helper_functions["Kurtosisses"](
+            X_transformed, y, logger, categorical_transformed)
+    )
+
+    if request.param == 'numpy':
+        return X_transformed, y, categorical_transformed
+    elif request.param == 'pandas':
+        return pd.DataFrame(X_transformed), y, categorical_transformed
+    else:
+        raise ValueError(request.param)
+
+
+class TestMetaFeatures:
```

Comment:
This looks all good, but I'm wondering whether there's a reason not to go all in with pytest and also ditch this class?

Reply:
No Reason at all. I have removed this class, but also moved the sparse metafeatures to pytest.